### PR TITLE
refactor(web): better centralizes OSK layout internals to prepare for optimization efforts 🪠

### PR DESCRIPTION
**Pull request by jahorton**
To prepare to optimize layout-reflow for Web's OSK, it will help us to better centralize functionality and properties related to the OSK's "layer group" element.  Doing so will help us to precalculate and cache values related to layout reflow without actually _triggering_ layout reflow repeatedly during use.  An initial reflow may be needed to obtain specific values, though.

This PR has been spun off of an older state of https://github.com/keymanapp/keyman/pull/11140. Certain design decisions may make more sense with the whole picture in mind.  In particular, the new `verticalPadding` property and its backing, cached field will help in reducing layout-reflow in followups.

@keymanapp-test-bot skip

Any testing relevant to do here may as well be done as part of #11140.

### CODE REVIEW

File: web/src/engine/osk/src/keyboard-layout/oskLayerGroup.ts [hunk 1]
```diff
@@ -249,8 +256,35 @@ export default class OSKLayerGroup {
...
+
+    // Assumption:  this styling value will not change once the keyboard and
+    // related stylesheets are loaded and applied.
+    if(this._heightPadding === undefined) {
+      // Should not trigger a new layout reflow; VisualKeyboard should have made
+      // no further DOM style changes since the last one.
+
+      // For touch-based OSK layouts, kmwosk.css may include top & bottom
+      // padding on the layer-group element.
+      const computedGroupStyle = getComputedStyle(this.element);
```
Comment (jahorton) [inline web/src/engine/osk/src/keyboard-layout/oskLayerGroup.ts:274]: Also of note:  `VisualKeyboard.refreshLayout()` and `.buildDocumentationKeyboard` will not call this method unless the OSK is in the DOM, so we don't need to check for that.

Only just now realized that I left that assumption unstated here.

**Review by ermshiperete — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.307-beta](https://keyman.com/downloads/releases/beta/17.0.307)

File: web/src/engine/osk/src/keyboard-layout/oskLayerGroup.ts (hunk 1 above)
Comment (mcdurdin) [reply web/src/engine/osk/src/keyboard-layout/oskLayerGroup.ts:274]: This comment above ^^^ should be in the code, not in a PR comment.

**Review by mcdurdin — COMMENTED** [review]
This looks fine

**Pull request merged**